input/postgres/relation_stats_ext.go

```diff
@@ -51,3 +51,80 @@ func handleRelationStatsExt(db *sql.DB, relStats state.PostgresRelationStatsMap,
 
 	return relStats, nil
 }
+
+const citusIndexSizeSQL = `
+WITH dist_idx_shard_stats_raw AS (
+	SELECT
+		dp.logicalrelid::regclass,
+		(pg_catalog.run_command_on_shards(logicalrelid::regclass::text, $$
+			SELECT
+				jsonb_agg(shard_idx_stats)
```

Comment:
Because we need to run this once for each table, and because `run_command_on_shards` can only return a scalar value, pack that into jsonb to get info for all indexes at once for each table.

Reply:
Makes sense, I think that works.

One thought that just crossed my mind: It may be interesting to consider switching this query to using `run_command_on_workers` in the future, and instead of doing one distributed query per table, doing a single distributed query for all tables, that returns the statistics from each worker node together.

But since our current implementation for table sizes already does one query per table (as `citus_table_size` does that under the hood), so this is matches that existing pattern, and I think its acceptable for most clusters.